pkg/resourceinterpreter/customized/declarative/luavm/lua.go

```diff
@@ -197,6 +197,35 @@ func (vm *VM) setLib(l *lua.LState) error {
 			return err
 		}
 	}
+
+	// Set potentially unsafe basic functions to nil, prohibiting users from
+	// calling these functions in custom Lua scripts.
+	// For the safety analysis of Sandbox in Lua, please refer to http://lua-users.org/wiki/SandBoxes
+	// For users, these functions are not needed for parsing resource templates,
+	// so disabling these functions has no impact on functionality.
+	for _, value := range []string{
+		"collectgarbage",
+		"dofile",
+		"getfenv",
+		"getmetatable",
+		"load",
+		"loadfile",
+		"loadstring",
+		"rawequal",
+		"rawget",
+		"rawset",
+		"setfenv",
+		"setmetatable",
+		"module",
+		"newproxy",
```

Comment:
Is this the full set of unsafe functions?

Reply:
There is also a `require` function. I have tested this function and found that it can only load the `kube` and `os` (can only call time and date functons) library, not other libraries. Therefore, `require` is safe.